### DISCUSSION TITLE
CR Touch not working properly

### DISCUSSION THREAD

**Discussion by clecroy1** [Jyers/Marlin#1992] [General Discussion]
I recently purchased a CR Touch, I downloaded https://github.com/Jyers/Marlin/releases/download/v2.0.1/E3V2-BLTouch-3x3-v4.2.7-v2.0.1.bin and the download seemed to go good. I then homed the machine and everything seemed to work correctly till the Z-axis. as soon as it got to the z axis it raised up and wont go down and just continues to probe. I then tried to "create a new mesh" and it wont ever touch the build plate. I tried to manually move the Z axis but it wont let me. I don't know what I'm doing wrong I'm sure it is a easy fix and I'm missing something. but I have searched the internet and cant find anything.

Thank you in advance I hope to be able to use the CR touch soon.

**Comment by darco415** [comment]
Confirm you have a 4.2.7 board. After flashing, in menu choose to restore defaults...

**Reply by clecroy1** [reply]
Yes correct! I have downloaded it multiple times trying different routes. But have not gotten any progress.

**Reply by darco415** [reply]
is your endstop disabled?

**Reply by clecroy1** [reply]
Are you talking about the z limit switch? If so I unplugged it because from what I understand the probe is now the new z limit when working properly.

**Reply by epistoteles** [reply]
Not sure if this helps for the precompiled binaries, but I had the exact same issue and setting the config parameters as described in the comments of this Reddit thread fixed it for me: https://www.reddit.com/r/ender5/comments/phifb1/cr_touch_marlin_20/

**Comment by Siggy101** [comment]
A few things to check:

I think you unplugged your z-limit switch.  Maybe double check that you unplugged it fully.  It does indeed need to be unplugged.  I removed the cable from the board end too.  Be sure to physically remove the limit switch from the chassis extrusion so nothing crashes.

Also check that nothing else was unplugged or knocked loose when you did the above step.  The fact that your z axis moves up but not down, even manually, makes me wonder about the physical connection.  I would unplug and re-seat both ends of the z-axis stepper motor cable.  You could also swap the cable with another one to be sure it's not damaged.

Are you 100% sure you have a v4.2.7 board?  Ender 3 V2s are delivered new with both v4.2.2 and v4.2.7 so even if your printer is brand new, you may have a v4.2.2 board (I have a v4.2.2 in my E3v2 purchased new in May 2022).  To check your board version, open up the access panel to the mainboard and the version is printed in the middle of the board.  If you have a v4.2.2 board then you need to grab the 4.2.2 build as it's different.

One point that's a bit more of a tip than a fix, why not grab a UBL build?  I started with the 'BLTouch_3x3' build but moved up to the UBL.  The levelling is absolutely first class.
Also, be sure to update the screen firmware to get those lovely colour graphics.

**Reply by clecroy1** [reply]
Hello I made sure that everything was connected properly. I also returned it to stock software and it works but of course with out the BL touch. The board is a v4.2.7 is there any programming that I am needing to do in the software that I didn't know about? I dont see a UBL 3x3 for the bl touch.

**Reply by Siggy101** [reply]
This is the build that you want:  https://github.com/Jyers/Marlin/releases/download/v2.0.1/E3V2-UBL-BLTouch-15x15-v4.2.7-v2.0.1.bin
Use the 15x15 to build the super detailed mesh.  Yes, it takes a long time to build the first mesh but then you put in your slicer start code to just probe 2x2 or maybe 3x3 (not really needed).  This means you get the benefits of a 225 point mesh but the speed of a 4 point probe to just tilt the mesh.
Give it a whirl and see if that build fixes your weird issue.  Then you can start playing with the joys of UBL.  It's a little learning curve but in a day or 2, you'll be printing amazing 1st layers.
For info, I use that build (but the 4.2.2. one) and I built my mesh at least 2 months ago.  After about 75 prints, just tilting the mesh with a 2x2 probe at the start of each print, I have never had 1st layer issues and have not had to to anything to the mesh in that time.

**Reply by Siggy101** [reply]
One tip:  when you build your initial mesh, be sure to heat your bed to your normal 1st layer temp for a good 15 or 20 minutes to allow it to be really, really evenly heated and all expansion to happen.
